### DISCUSSION TITLE
Cross-language support

### DISCUSSION THREAD

**Discussion by janvhs** [pop-os/libcosmic#105] [General]
Hey, I wasn't able to find anything online, but are there any thoughts on adding a possibility for a usage from languages like Go, Python or Zig in the future?

Although Rust is an awesome language, it is still hard to learn and a lot of software is written in languages like Python, Go or even C. Therefore, software written in these languages are currently not able to use this toolkit. It would be great to have a wide verity of software written in a desktop environment native fashion.
A toolkit which currently uses a polyglot approach is GTK and Libadwaita. Although these libraries are written in C they can be used from multiple languages. Is there a future like this for the COSMIC-desktop and if not are there any plans to prevent software which looks _out of place_ for this desktop environment?

**Comment by idoric** [comment]
@janvhs Maybe you'll be interested, I found a project for a [Python wrapper for Rust Iced](https://github.com/icedpygui/IcedPyGui). One can imagine that it could be extended to handle libcosmic.

**Reply by janvhs** [reply]
Uhhh thank you!
